src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -215,15 +203,17 @@ protected override void StartWorking()
             }
 
             OnSourceProviderSourcesChanged(this, EventArgs.Empty);
-            OnWorkspaceChanged(localSolutionChanged: true, localChangedProject: null);
+
+            // Kick off an initial set of work that will analyze the entire solution.
+            _workQueue.AddWork((solutionChanged: true, changedProject: null));
         }
 
         private void OnSourceProviderSourcesChanged(object sender, EventArgs e)
         {
             lock (_gate)
             {
-                // Reset the task for loading package sources.
-                _packageSourcesAsync = null;
+                // Reset the task for loading package sources.  The next time someone asks for it we'll recompute it.
```

Comment:
that's a fair question.  we could potentially do that.  but we'd have to audit to make sure these events don't get fired early, causing us to load the package sources too early.

Reply:
I guess I would imagine the logic would be:

1. If it's null, nobody asked so keep the task null.
2. If it's non-null then somebody did want the result and so we should fire a new task to refresh it.

Otherwise this seems like we might be setting up for the less-than-great experience that somebody is updating their sources to include nuget.org so they can use this feature, a gesture which then prevents the feature from working for a bit. :smile:

Reply:
yes.  that's a much better approach.

Reply:
done.